src/components/Textfield/Textfield.tsx

```diff
@@ -0,0 +1,252 @@
+import type { ChangeEventHandler, MouseEventHandler } from 'react';
+import {
+  forwardRef,
+  useCallback,
+  useEffect,
+  useImperativeHandle,
+  useMemo,
+  useRef
+} from 'react';
+
+import Icon from '@/components/Icon';
+import FormItem from '@/components/Shared/FormItem';
+import type { ThemeType } from '@/components/Shared/FormItem/types';
+import Spinner from '@/components/Spinner';
+import useDebounce from '@/hooks/useDebounce';
+
+import { cx } from '@/utils/css';
+import { noop } from '@/utils/misc';
+import { DEFAULT_DEBOUNCE_DELAY } from '@/constant/input';
+import { GRAYMAUVE1000, VIOLET900 } from '@/constant/theme';
+
+import * as styles from './style.module.scss';
+import type { TextfieldProps } from './types';
+
+const Textfield = forwardRef<HTMLInputElement, TextfieldProps>(
+  (props: TextfieldProps, ref) => {
+    const {
+      className,
+      debounceDelay = DEFAULT_DEBOUNCE_DELAY,
+      disabled = false,
+      disabledDebounce = false,
+      enableClear = false,
+      error = false,
+      helper,
+      label,
+      loading,
+      maxLength,
+      onChange = noop,
+      optional,
+      placeholder = 'Value input',
+      prefix,
+      required,
+      rules,
+      showCounter,
+      success,
+      suffix,
+      value: propsValue = '',
+      width = '300px',
+      ...res
+    } = props;
+    const containerRef = useRef<HTMLElement>(null);
+    const inputRef = useRef<HTMLInputElement>(null);
+
+    const handleOnChangeDebounceValue = useCallback(
+      (val: string) => {
+        onChange(val, inputRef.current ?? undefined);
+      },
+      [onChange]
+    );
+
+    const [value = '', setValue] = useDebounce<string>(
+      propsValue,
+      handleOnChangeDebounceValue,
+      disabledDebounce ? 0 : debounceDelay
+    );
+
+    useImperativeHandle(ref, () => {
+      return inputRef.current as HTMLInputElement;
+    });
+
+    const handleOnChange: ChangeEventHandler<HTMLInputElement> = useCallback(
+      (e) => {
+        // INFO: special case if user clear the textfield still trigger onChange props
+        if (e.target.value === '') setValue(e.target.value);
+
+        // INFO: skip if regex test result is false
+        if (rules && !rules.test(e.target.value)) {
+          return;
+        }
+
+        setValue(e.target.value);
+      },
+      [rules, setValue]
+    );
+
+    const handleOnClickClearBtn: MouseEventHandler<HTMLSpanElement> =
+      useCallback(
+        (e) => {
+          e.preventDefault();
+          setValue('');
+        },
+        [setValue]
+      );
+
+    const helperTheme = useMemo(() => {
+      let theme: ThemeType = 'initial';
+
+      if (error) theme = 'error';
+      if (success) theme = 'success';
+      if (disabled) theme = 'disabled';
```

Comment:
pake else if aja soalnya mending pilih salah satu aja theme nya
```suggestion
      if (error) theme = 'error';
      else if (success) theme = 'success';
      else if (disabled) theme = 'disabled';
```